src/com/goide/inspections/unresolved/GoIntroduceFunctionFix.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2013-2016 Sergey Ignatov, Alexander Zolotov, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections.unresolved;
+
+import com.goide.psi.*;
+import com.goide.psi.impl.GoElementFactory;
+import com.goide.psi.impl.GoTypeUtil;
+import com.goide.refactor.GoRefactoringUtil;
+import com.intellij.codeInsight.CodeInsightUtilCore;
+import com.intellij.codeInsight.template.*;
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement;
+import com.intellij.diagnostic.AttachmentFactory;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.command.CommandProcessor;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.RangeMarker;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.*;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GoIntroduceFunctionFix extends LocalQuickFixAndIntentionActionOnPsiElement {
+  private final String myName;
+  private static final String myFamilyName = "Create function";
+
+  public GoIntroduceFunctionFix(@NotNull PsiElement element, @NotNull String name) {
+    super(element);
+    myName = name;
+  }
+
+  @Override
+  public void invoke(@NotNull Project project,
+                     @NotNull PsiFile file,
+                     @Nullable("is null when called from inspection") Editor editor,
+                     @NotNull PsiElement startElement,
+                     @NotNull PsiElement endElement) {
+    if (editor == null) {
+      LOG.error("Cannot run quick fix without editor: " + getClass().getSimpleName(),
+                AttachmentFactory.createAttachment(file.getVirtualFile()));
+      return;
+    }
+
+    if (!(startElement instanceof GoCallExpr)) {
+      return;
+    }
+
+    GoCallExpr call = (GoCallExpr)startElement;
+    List<GoExpression> args = call.getArgumentList().getExpressionList();
+    ArrayList<GoType> argTypes = new ArrayList<>(args.size());
+    List<GoType> resultTypes = new ArrayList<>();
+    for (GoExpression arg : args) {
+      argTypes.add(arg.getGoType(null));
```

Comment:
you're processing all arguments in order to get their types in order to process these types with one more foreach in `createFunctionDeclaration` method?